pkg/solana/client.go

```diff
@@ -52,8 +53,16 @@ func NewClient(spec OCR2Spec, logger Logger) *Client {
 		ctxInterval = 2 * pollInterval
 	}
 
+	// parse tx context, if errors use defaultStaleTimeout
```

Comment:
a broader nit: right now the validation is split between core and chainlink-solana, I think we should just move it all into core since the earlier we get the raw strings into types the better. OCR2Spec can just be assumed to be a validated spec on the relay side

Reply:
I'm also not fond of the split but would like to have most network-specific code here instead of in Core. That doesn't mean that we can not execute earlier in Core, but execute code that is defined here.

On the other hand, for the short term, all/most of these args should probably be moved as Solana chain configuration in core to follow Ethereum & Terra.

Reply:
i don't mind either way (handling the validation in core or external to it)

And i agree that this needs to be refactored into a chain config, but that's a bit out of scope for this PR. I book marked it here to revisit later on: #176

Reply:
Of course, I didn't mean this PR but as a next first step to bring it up to date with Terra :)